crates/lox-io/src/ndm/common.rs

```diff
@@ -0,0 +1,1624 @@
+    pub impact_3_cross_track: Option<DistanceType>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RdmSpacecraftParametersType {
+    #[serde(rename = "COMMENT")]
+    pub comment_list: Vec<String>,
+    #[serde(rename = "WET_MASS")]
+    pub wet_mass: Option<MassType>,
+    #[serde(rename = "DRY_MASS")]
+    pub dry_mass: Option<MassType>,
+    #[serde(rename = "HAZARDOUS_SUBSTANCES")]
+    pub hazardous_substances: Option<String>,
+    #[serde(rename = "SOLAR_RAD_AREA")]
+    pub solar_rad_area: Option<AreaType>,
+    #[serde(rename = "SOLAR_RAD_COEFF")]
+    pub solar_rad_coeff: Option<NonNegativeDouble>,
+    #[serde(rename = "DRAG_AREA")]
+    pub drag_area: Option<AreaType>,
+    #[serde(rename = "DRAG_COEFF")]
+    pub drag_coeff: Option<NonNegativeDouble>,
+    #[serde(rename = "RCS")]
+    pub rcs: Option<AreaType>,
+    #[serde(rename = "BALLISTIC_COEFF")]
+    pub ballistic_coeff: Option<BallisticCoeffType>,
+    #[serde(rename = "THRUST_ACCELERATION")]
+    pub thrust_acceleration: Option<Ms2Type>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct AltType {
+    #[serde(rename = "$text")]
+    pub base: AltRange,
+    #[serde(rename = "@units")]
+    pub units: Option<LengthUnits>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct BallisticCoeffType {
+    #[serde(rename = "$text")]
+    pub base: NonNegativeDouble,
+    #[serde(rename = "@units")]
+    pub units: Option<BallisticCoeffUnitsType>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct LatType {
+    #[serde(rename = "$text")]
+    pub base: LatRange,
+    #[serde(rename = "@units")]
+    pub units: LatLonUnits,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct LonType {
+    #[serde(rename = "$text")]
+    pub base: LonRange,
+    #[serde(rename = "@units")]
+    pub units: LatLonUnits,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedType {
+    #[serde(rename = "COMMENT")]
+    pub comment_list: Vec<String>,
+    #[serde(rename = "USER_DEFINED")]
+    pub user_defined_list: Vec<UserDefinedParameterType>,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedParameterType {
+    #[serde(rename = "$text")]
+    pub base: String,
+    #[serde(rename = "@parameter")]
+    pub parameter: String,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionType {}
```

Comment:
Are the quarternion types intentionally empty?

Reply:
Thanks for spotting this.

I missed this. It's empty because of https://github.com/media-io/xml-schema/pull/36. Other choice definitions I implemented manually.

This is unused by the messages I implemented. This is because I implemented [CCSDS 502.0-B-3](https://public.ccsds.org/Pubs/502x0b3e1.pdf), but now I realize that there's a whole other standard [CCSDS 505.0-B-3](https://public.ccsds.org/Pubs/505x0b3e1.pdf) and these use this type.

I am fine with dropping this for now and then following up with implementation for the other messages.